src/js/game/systems/item_ejector.js

```diff
@@ -159,21 +159,26 @@ export class ItemEjectorSystem extends GameSystemWithFilter {
                     continue;
                 }
 
+                const maxProgress = Math.min(
+                    1.0,
+                    sourceSlot.cachedBeltPath ? sourceSlot.cachedBeltPath.spacingToFirstItem * 1.45 : 1.0
+                );
+
                 // Advance items on the slot
                 sourceSlot.progress = Math.min(
-                    1,
+                    maxProgress,
                     sourceSlot.progress +
                         progressGrowth *
                             this.root.hubGoals.getBeltBaseSpeed() *
                             globalConfig.itemSpacingOnBelts
                 );
 
                 if (G_IS_DEV && globalConfig.debug.disableEjectorProcessing) {
-                    sourceSlot.progress = 1.0;
+                    sourceSlot.progress = maxProgress;
                 }
 
                 // Check if we are still in the process of ejecting, can't proceed then
-                if (sourceSlot.progress < 1.0) {
+                if (sourceSlot.progress < maxProgress) {

```

Comment:
Shouldn't this still use 1.0 instead of `maxProgress`?

If I understand correctly, `maxProgress` just says how far the ejection process can get, it doesn't change when that process should be considered complete, and this check is meant to ensure that the process is complete before we hand the item off to the belt (or next building).

I think it will still work as-is, since `tryAcceptItem` below will probably return `false` (the belt not being ready for new items), but it seems like we shouldn't be trying to do that yet, and it will probably also be slower due to the extra processing.